wire-ios-request-strategy/Sources/Request Strategies/User/UserProfileRequestStrategy.swift

```diff
@@ -58,12 +62,11 @@ public class UserProfileRequestStrategy: AbstractRequestStrategy, IdentifierObje
 
     public override func nextRequestIfAllowed(for apiVersion: APIVersion) -> ZMTransportRequest? {
         fetchAllConnectedUsers(for: apiVersion)
-        switch apiVersion {
-        case .v0:
-            return userProfileByID.nextRequest(for: apiVersion)
-        case .v1, .v2, .v3, .v4, .v5:
-            return userProfileByQualifiedID.nextRequest(for: apiVersion)
-        }
+
+        return
+            userProfileByID.nextRequest(for: apiVersion) ??
+            userProfileByQualifiedID.nextRequest(for: apiVersion) ??
+            actionSync.nextRequest(for: apiVersion)
     }
```

Comment:
@johnxnguyen Is this something that we can do regarding the requests? I would like the requests to execute one after the other or to make sure that `actionSync.nextRequest(for: apiVersion)` is being executed

Reply:
Yes but should should make sure that `userProfileByID` returns a `nil` when the api version is 1 or greater.

Reply:
@johnxnguyen So something like this?

```swift
public override func nextRequestIfAllowed(for apiVersion: APIVersion) -> ZMTransportRequest? {
    fetchAllConnectedUsers(for: apiVersion)

    // Check the apiVersion before userProfileByID.nextRequest
    if apiVersion.rawValue >= 1 {
        return
            userProfileByQualifiedID.nextRequest(for: apiVersion) ??
            actionSync.nextRequest(for: apiVersion)
    } else {
        return
            userProfileByID.nextRequest(for: apiVersion) ??
            userProfileByQualifiedID.nextRequest(for: apiVersion) ??
            actionSync.nextRequest(for: apiVersion)
    }
}
```

Reply:
I was thinking more like the `userProfileByIDTranscoder` returning nil if the api version is greater than zero.

Reply:
Fixed it, thank you!